examples/doc/over_Z.g

```diff
@@ -0,0 +1,39 @@
+#! @BeginChunk over_Z
+LoadPackage( "Bialgebroid" );
+
+#! We create two algebroids from the same algebra.
```

Comment:
How about "from the same quiver"? Technically they are created using the same algebra, but mathematically this seems confusing.

Reply:
Only writing "quiver" would not be enough, since I also specify relations.

I could write "quiver and relations", but this would, in my opinion, raise false expectations.
Let me try to explain why this is the case:

Mathematically, the best approach to algebroids over Z would be the following:
- define a quiver q
- define the path algebra over the integers Z, denoted by Zq
- specify a set of relations R, i.e., elemens of Zq
Then, use the pair (q, R) for the definition of the algebroid over Z, call it Alg_Z(q,R).

Unfortunately, the only mathematically entities I can work with are:
- a quiver q
- the path algebra over the rationals Q, denoted by Qq
- sets of relations R specified by elements in Qq
So, how can I construct a mathematically valid algebroid over Z using these data?
I proceed as follows:
1) construct the path algebra A = Qq/R over Q
2) let QPA compute a Q-basis of its paths
3) check if, w.r.t. this Q-basis, the coefficients in the multiplication table T are integers, and if yes, then use the Q-basis as a Z-basis for my algebroid B, for which I also use T as a multiplication table.
(See also my documentation of the construction `AlgebroidOverZ`).

But note that the algebroid B is mathematically NOT the same object as Alg_Z(q,R) (assuming the elements in R only consists of integral linear combinations). A simple counter-example:

q: quiver with two nodes v,w and one edge v->w denoted by alpha
R: 2*alpha

Then, Hom(v,w) considered in Alg_Z(q,R) is non-trivial, but has torsion. However, by construction, B cannot have torsion. In this example, in B we would indeed have Hom(v,w) = 0.

So, I really would like to stick to the term that the algebroid was created using the algebra, and not using the quiver or using the quiver and relations.

Reply:
Thank you for the explanation. Then I agree that the algebra is essential here and I think that this line should remain as is.

It seems that you are basically descending B from Q to Z. Is that correct?

Reply:
Yes, it is correct in the following sense: tensoring B with Q over Z gives back the algebra Qq/R.

Reply:
This is what I meant.

I think it could be interesting to make this construction available in the form of taking an algebroid over Q as input and returning the descended algebroid over Z (if the described descent approach works) and maybe `fail` otherwise. This could be called something like `DescentToZ`. I think this would be preferable as algebroids over Q that are actually defined over Z may arise differently. Then it would be nice to be able to descent them too instead of creating them using `AlgebroidOverZ` using the underlying path algebra (with relations). What do you think?

Reply:
One may even consider to pass the "destination ring" (here Z) as a second argument, allowing this to be extended in the future to other rings.

Reply:
I like the idea. I think it makes the mathematics behind the procedure more understandable.

If you are okay with its technical downside, namely that constructing an algebroid over Q always needs to have the underlying algebra over Q constructed first, I would proceed as follows for its realization:  
- I would create the constructor `DecentToZDefinedByBasisPaths` for `IsAlgebroid`. The Suffix `DefinedByBasisPaths` shall be a reminder of the fact that descents are not unique and that this particular one is created using the basis paths of the input algebra
- This constructor would access the underlying path algebra and then delegate to the constructor that I've written in this PR.

I'm not so sure about the "destination ring" (call it D). If R is the ring over which the algebroid is originally defined, then the input should rather be a D-algebra structure for R, i.e., a ring homomorphism phi: D -> R.

In the particular case for Z in Q, I use the fact that the GAP integers are embedded in GAP's rationals on a technical level: I can multiply integers with rationals without conversions, and ask if a rational is an integer via the filter `IsInt`. This is of course a very convenient situation that I don't have in general. So, if you're okay with it, I would like to leave it at the case Z in Q.

Reply:
> If you are okay with its technical downside, namely that constructing an algebroid over Q always needs to have the underlying algebra over Q constructed first
What do mean exactly? All constructors that we have now (without this PR) would remain, is not it?

In order to construct an algebroid over Z one would have to construct one over Q first, i.e. what is now `AlgebroidOverZ(Qq)` would become `DecentToZDefinedByBasisPaths(Algebroid(Qq))`, right? I think this would actually be good as it makes the mathematics more clear.

As it stands, `AlgebroidOverZ` could easily be thought of as doing the following (when ignoring its documentation):

> Mathematically, the best approach to algebroids over Z would be the following:
> 
>     define a quiver q
>     define the path algebra over the integers Z, denoted by Zq
>     specify a set of relations R, i.e., elemens of Zq
>     Then, use the pair (q, R) for the definition of the algebroid over Z, call it Alg_Z(q,R).

> I'm not so sure about the "destination ring" (call it D). If R is the ring over which the algebroid is originally defined, then the input should rather be a D-algebra structure for R, i.e., a ring homomorphism phi: D -> R.
> 
> In the particular case for Z in Q, I use the fact that the GAP integers are embedded in GAP's rationals on a technical level: I can multiply integers with rationals without conversions, and ask if a rational is an integer via the filter IsInt. This is of course a very convenient situation that I don't have in general. So, if you're okay with it, I would like to leave it at the case Z in Q.

It was just an idea. I am fine with it being implemented for the descent from Q to Z only, just as you propose it.

Reply:
I meant:
- with the current version, B is technically constructed without constructing the GAP object `Algebroid( Qq/R )`.
- with the suggested, mathematically more understandable version, for constructing B, one needs to construct the GAP object `Algebroid( Qq/R )` first.

So, the mathematically more understandable version comes at a price: the creation of an auxiliary object (namely `Algebroid( Qq/R )`) that is never used within the creation of B, since all I need from `Algebroid( Qq/R )` is the path algebra Qq/R.

But still, I'm willing to do this.

Reply:
I would value mathematical clarity over performance. I am happy with the new version. Thank you very much.